zerver/actions/streams.py

```diff
@@ -470,14 +470,13 @@ def send_subscription_add_events(
 
         # Send a notification to the user who subscribed.
         event = dict(type="subscription", op="add", subscriptions=sub_dicts)
-        send_event(realm, event, [user_id])
+        send_event_on_commit(realm, event, [user_id])
 
 
 # This function contains all the database changes as part of
-# subscribing users to streams; we use a transaction to ensure that
-# the RealmAuditLog entries are created atomically with the
-# Subscription object creation (and updates).
-@transaction.atomic(savepoint=False)
+# subscribing users to streams; the caller of this function
+# use a transaction to ensure that the RealmAuditLog entries
+# are created atomically with the Subscription object creation (and updates).
 def bulk_add_subs_to_db_with_logging(
```

Comment:
Is there any harm in having  `@transaction.atomic(savepoint=False)` on this? That usage model is nestable.

Reply:
I was just trying to avoid nested transactions as they are not of any help in these cases? Let me know if they help in some way.

Reply:
`@transaction.atomic(savepoint=False)` is a noop if we're already in a transaction -- that's effectively what `savepoint=False` does.

Reply:
Yeah, my thought was to remove as it is a noop.

But your reasoning makes sense.